bin/e2hs-writer/src/reader.rs

```diff
@@ -0,0 +1,381 @@
+                let header_validator = HeaderValidator::new();
+                Some(Arc::new(
+                    lookup_epoch_acc(
+                        epoch_index,
+                        &header_validator.pre_merge_acc,
+                        &epoch_acc_path,
+                    )
+                    .await?,
+                ))
+            }
+            false => None,
+        };
+        Ok(Self {
+            starting_block,
+            ending_block,
+            epoch_accumulator,
+            execution_api,
+            era_provider,
+        })
+    }
+
+    fn get_pre_merge_block_data(&self, block_number: u64) -> anyhow::Result<AllBlockData> {
+        let raw_era1 = self.era_provider.get_era1_for_block(block_number)?;
+        let block_index = block_number % EPOCH_SIZE;
+        let tuple = Era1::get_tuple_by_index(&raw_era1, block_index);
+        let header = tuple.header.header;
+        let receipts = tuple.receipts.receipts;
+        let body = tuple.body.body;
+        let epoch_acc = self.epoch_accumulator.clone().unwrap_or_else(|| {
+            panic!("Epoch accumulator not found for pre-merge block: {block_number}")
+        });
+        let proof = PreMergeAccumulator::construct_proof(&header, &epoch_acc)?;
+        let proof = BlockProofHistoricalHashesAccumulator::new(proof.into()).map_err(|e| {
+            anyhow!("Unable to convert proof to BlockProofHistoricalHashesAccumulator: {e:?}")
+        })?;
+        let header_with_proof = HeaderWithProof {
+            header,
+            proof: BlockHeaderProof::HistoricalHashes(proof),
+        };
+        Ok(AllBlockData {
+            block: block_number,
+            header_with_proof,
+            body,
+            receipts,
+        })
+    }
+
+    async fn get_pre_capella_block_data(&self, block_number: u64) -> anyhow::Result<AllBlockData> {
+        let era = self.era_provider.get_era_for_block(block_number)?;
+        let block = era
+            .blocks
+            .iter()
+            .find(|block| block.block.execution_block_number() == block_number)
+            .unwrap_or_else(|| {
+                panic!("Era file for block #{block_number} not found during pre-capella lookup")
+            });
+        let block = block
+            .block
+            .message_merge()
+            .map_err(|e| anyhow!("Unable to decode merge block: {e:?}"))?;
+        let payload = block.body.execution_payload.clone();
+        let header = pre_capella_execution_payload_to_header(payload.clone());
+        let historical_batch = HistoricalBatch {
+            state_roots: era.era_state.state.state_roots().clone(),
+            block_roots: era.era_state.state.block_roots().clone(),
+        };
+        let slot = block.slot;
+        // beacon block proof
+        let hb_proof = historical_batch.build_block_root_proof(slot % EPOCH_SIZE);
+        let beacon_block_proof: FixedVector<B256, typenum::U14> = hb_proof.into();
```

Comment:
```suggestion

        // beacon block proof
        let beacon_block_proof = historical_batch.build_block_root_proof(slot % EPOCH_SIZE);
        let beacon_block_proof: FixedVector<B256, typenum::U14> = beacon_block_proof.into();
        
```

We can add some strategic spaces for readability and code sectioning and I don't know what hb means, so if we could expand on that it would be cool